src/blueprint.zig

```diff
@@ -105,9 +108,47 @@ pub const Blueprint = struct {
 		}
 		return .{.success = self};
 	}
+
+	pub const PasteMode = enum {all, degradable};
+
+	pub fn pasteInGeneration(self: Blueprint, pos: Vec3i, chunk: *ServerChunk, mode: PasteMode) void {
+		switch(mode) {
+			.all => _pasteInGeneration(self, pos, chunk, .all),
+			.degradable => _pasteInGeneration(self, pos, chunk, .degradable),
+		}
+	}
+
+	fn _pasteInGeneration(self: Blueprint, pos: Vec3i, chunk: *ServerChunk, comptime mode: PasteMode) void {
+		const startX = pos[0];
+		const startY = pos[1];
+		const startZ = pos[2];
+
+		for(0..self.blocks.width) |x| {
+			const worldX = startX + @as(i32, @intCast(x));
+
+			for(0..self.blocks.depth) |y| {
+				const worldY = startY + @as(i32, @intCast(y));
+
+				for(0..self.blocks.height) |z| {
+					const worldZ = startZ + @as(i32, @intCast(z));
+					if(!chunk.liesInChunk(worldX, worldY, worldZ)) continue;
```

Comment:
`liesInChunk` expects relative coordinates, but you give it world coordinates?
I think you got your names wrong.